rclrs/src/logging/log_params.rs

```diff
@@ -0,0 +1,293 @@
+use crate::{rcl_bindings::RCUTILS_LOG_SEVERITY, Clock};
+use std::{borrow::Borrow, ffi::CString, time::Duration};
+
+/// These parameters determine the behavior of an instance of logging.
+#[derive(Debug, Clone, Copy)]
+pub struct LogParams<'a> {
+    /// The name of the logger
+    logger_name: LoggerName<'a>,
+    /// The severity of the logging instance.
+    severity: LogSeverity,
+    /// Specify when a log message should be published (See[`LoggingOccurrence`] above)
+    occurs: LogOccurrence,
+    /// Specify a publication throttling interval for the message.  A value of ZERO (0) indicates that the
+    /// message should not be throttled. Otherwise, the message will only be published once the specified
+    /// interval has elapsed. This field is typically used to limit the output from high-frequency messages,
+    /// e.g. if `log!(logger.throttle(Duration::from_secs(1)), "message");` is called every 10ms, it will
+    /// nevertheless only be published once per second.
+    throttle: Duration,
```

Comment:
Nit, can this be an `Option<Duration>` instead to be a bit more idiomatic about whether a throttling setting exists or not?

Reply:
I did consider this, but I decided to stick with simply `Duration` for the following reasons:

* This field is private, so whether it's `Duration` or `Option<Duration>` is an implementation detail that should be considered invisible to users.
* Users never need to know to set `Duration::new(0, 0)` to get no throttle because no throttle is implied by default. Users only need to specify positive throttle. Therefore as far as users are concerned, there is no special rule that they need to know about.
* We need to check for zero or negative throttle anyway, even if we have an `Option<Duration>` because a negative duration would be invalid and a zero duration should have us skip throttling, just like we currently do, so there's no advantage for the implementation.
* A zero duration throttle is mathematically equivalent to no throttle, so we're not introducing a nonsense meaning for the value, in contrast to a simulator using 0 mass to mean an object is unmovable, where 0 is being given the opposite of its actual physical meaning.
* This is a very very minor, but `Option<Duration>` has a slightly larger memory footprint than `Duration`.

I'm open to reconsidering, but overall I would still lean towards just using `Duration`.